go/state/state_db_test.go

```diff
@@ -4143,6 +4143,70 @@ func TestSlotIdOrder(t *testing.T) {
 	}
 }
 
+func TestStateDB_SetTransientState_NewValue_UndoRemovesValue(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mock := NewMockState(ctrl)
+	db := createStateDBWith(mock, defaultStoredDataCacheSize, true)
+
+	db.SetTransientState(address1, key1, val1)
+
+	// Call undo
+	for _, f := range db.undo {
```

Comment:
here and in other tests do not explicitly execute the undo method. Instead test that proper semantics of when undo is triggered. In particular, create a snapshot, insert values, tests that values are available. Perform rollback and test that previous variables are available. Do this in 2-3 nested calls, i.e. a few snapshots are created, rolled back, etc. Also include Start/End of transaction that empties the storage